core/app/mailers/spree/carton_mailer.rb

```diff
@@ -6,7 +6,7 @@ def shipped_email(carton_id, resend: false)
       @carton = Spree::Carton.find(carton_id)
       subject = (resend ? "[#{Spree.t(:resend).upcase}] " : '')
       subject += "#{Spree::Store.current.name} #{Spree.t('shipment_mailer.shipped_email.subject')} ##{@carton.order_numbers.join(', ')}"
-      mail(to: @carton.order_emails, from: from_address, subject: subject)
+      mail(to: @carton.order_emails, from: from_address(Spree::Store.current), subject: subject)
```

Comment:
This is fine (and an improvement) for now, but we should replace this with the correct store for the order, rather than the default store.